libvast/test/experimental_table_slice.cpp

```diff
@@ -43,6 +44,17 @@ auto make_slice(const record_type& layout, Ts&&... xs) {
   return slice;
 }
 
+template <class T, class... Ts>
+auto make_legacy_slice(const record_type& layout, std::vector<T> x0,
+                       std::vector<Ts>... xs) {
+  auto builder = arrow_table_slice_builder::make(type{"rec", layout});
+  for (size_t i = 0; i < x0.size(); ++i) {
+    CHECK(builder->add(x0.at(i)));
+    CHECK(builder->add(xs.at(i)...));
+  }
+  return builder->finish();
+}
```

Comment:
This will not work anymore son. Can you maybe create a record batch directly?

Reply:
Hm, that's a bummer - of course you're right. On the other hand, the entire idea of the test is to construct a valid old table slice, and I don't really feel like re-implementing half of the "old" table slice builder logic to create proper outdated record batches.

An alternative would be to write the table slice into some sort of fixture, which would make the test considerably easier to implement. I _think_ that's fine because we already lost the ability to write old table slices by then, so the only source for them is some previously written partition anyway.